app_feup/lib/view/Widgets/schedule_card.dart

```diff
@@ -63,18 +63,15 @@ class ScheduleCard extends GenericCard {
     final List<Widget> rows = <Widget>[];
 
     final now = DateTime.now();
-    var added = 0; // Lectures added to widget
-    var lastDayAdded = 0; // Day of last added lecture
-    final stringTimeNow = (now.weekday - 1).toString().padLeft(2, '0') +
-        now.toTimeHourMinString(); // String with current time within the week
-
+    var added = 0;  // Lectures added to widget
+    var lastDayAdded = 0;  // Day of last added lecture
+    
     for (int i = 0; added < 2 && i < lectures.length; i++) {
-      final stringEndTimeLecture = lectures[i].day.toString().padLeft(2, '0') +
-          lectures[i].endTime; // String with end time of lecture
-
-      if (stringTimeNow.compareTo(stringEndTimeLecture) < 0) {
-        if (now.weekday - 1 != lectures[i].day &&
-            lastDayAdded < lectures[i].day) {
+      if (now.hour < lectures[i].endTime.hour
+          || (now.hour == lectures[i].endTime.hour
+          && now.minute < lectures[i].endTime.minute)) {
```

Comment:
I think it is ok as it is, because either the hours can be inferior and minutes be random, or hours are the same but minutes are inferior... do you agree?